landing-zones/basic/data.tf

```diff
@@ -0,0 +1,13 @@
+data "azurerm_virtual_hub" "connectivity_hub" {
+  count               = var.enable_virtual_hub_connection == true ? 1 : 0
```

Comment:
Instead of doing a conditional like this, maybe we should make use of the `try()` function for cleaner looking code.

`try(var.enable_virtual_hub_connection, 0)`

Given that the variable is supposed to be a boolean (which is just an integer in the end), it should technically work.

Food for thought,

Reply:
hmm I'm not sure which is the better usage in this case (we do this all over to feature flag)

if True is just an integer, we probably also could at least simply the ternary

Reply:
True, if a boolean in Terraform is like in any other language, then just doing : `count = var.enable_virtual_hub_connection` should work just fine.

Reply:
Scratch what I said, apparently `terraform` has a different way of handling `bool` types.